pkg/referrer/unpack.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package referrer
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd/archive/compression"
+)
+
+// unpackFromTar unpacks the file named `source` in tar stream
+// and write into `target` path.
+func unpackFromTar(reader io.Reader, source, target string) error {
```

Comment:
It's better to move this functionality to a package handling OCI artifacts.

Reply:
Fixed, moved to `pkg/remote`.